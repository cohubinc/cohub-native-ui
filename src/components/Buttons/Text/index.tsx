import React, { PureComponent } from "react";
import Blank from "../Blank";
import Color from "src/definitions/enums/Color";
import Typography from "src/components/Typography";
import { TBlankButtonProps } from "src/components/Buttons/Blank";
import { StyleProp, TextStyle } from "react-native";

interface IProps {
  color?: Color;
  fontSize?: number;
  textStyle?: StyleProp<TextStyle>;
  block?: boolean;
}

export type TTextButtonProps = IProps & Omit<TBlankButtonProps, "color">;

export default class Text extends PureComponent<TTextButtonProps> {
  static defaultProps: Partial<IProps> = {
    color: Color.iconGrey,
    fontSize: 12
  };

  render() {
    const {
      color,
      fontSize,
      children,
      textStyle,
      block,
      style,
      className = "",
      disabled,
      ...rest
    } = this.props;

    return (
      <Blank
        {...rest}
        disabled={disabled}
        className={`${className} p-05`}
        style={{ display: block ? "block" : undefined, ...style }}
      >
        <Typography
          uppercase
          color={disabled ? Color.grey600 : color}
          weight={"500"}
          style={[{ fontSize }, textStyle]}
          kerning={0.07}
        >
          {children}
        </Typography>
      </Blank>
    );
  }
}
