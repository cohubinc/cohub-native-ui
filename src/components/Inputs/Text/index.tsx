import React from "react";
import { FieldRenderProps } from "react-final-form";

import TInputElementProps from "../definitions/TInputElementProps";
import Base from "../Base";

type FieldProps = FieldRenderProps<string, HTMLInputElement>;

interface IProps {
  label?: string;
  appearance?: "contrast" | "inverted";
  msgPosition?: {
    bottom: number;
  };
  "data-qa"?: string;
  input?: Partial<FieldProps["input"]>;
  meta?: FieldProps["meta"];
}

type TProps = IProps & Omit<TInputElementProps, "onChange" | "value">;

export default function Text(props: TProps) {
  const {
    input,
    meta,
    style,
    className,
    msgPosition = {},
    "data-qa": dataQa,
    ...restProps
  } = props;

  const { touched, error } = meta || ({} as any);

  const showError = !!(touched && error);

  return (
    <div {...{ className, style: { width: "100%", ...style } }}>
      <Base {...input} {...restProps} error={showError} data-qa={dataQa} />
    </div>
  );
}
