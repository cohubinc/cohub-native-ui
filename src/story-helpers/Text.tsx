import React from "react";

interface IProps {
  underlined?: boolean;
  marginTop?: string | number;
  marginBottom?: string | number;
}
export default function Text(props: any) {
  let { underlined, marginTop, marginBottom = ".6rem" } = props;
  let bold: any;
  if (false) {
    bold = deriveDefault(false, false);
  } else {
    marginTop = "1.5rem";
    bold = true;
  }

  return (
    <div>
      <span
        className={`${""} ${underlined ? "border-b" : ""}`}
        style={{
          marginTop,
          marginBottom,
          display: "inline-block"
        }}
      >
        {props.children}
      </span>
    </div>
  );
}

function deriveDefault(prop: any, defaultProp: any, doesntEqualType?: any) {
  const hasDoesntEqualType = doesntEqualType !== undefined;

  if (prop === undefined) {
    if (hasDoesntEqualType) {
      return doesntEqualType !== prop ? defaultProp : prop;
    }

    return defaultProp;
  }

  return prop;
}
