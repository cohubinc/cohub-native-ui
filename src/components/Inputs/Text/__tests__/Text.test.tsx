import React from "react";
import { mount } from "enzyme";
import Text from "../";

describe("Text Input", () => {
  it("renders without crashing", async () => {
    const wrapper = mount(<Text />);

    expect(wrapper).toMatchSnapshot();
  });
});
