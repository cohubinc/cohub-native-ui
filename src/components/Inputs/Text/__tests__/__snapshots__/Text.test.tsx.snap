// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text Input renders without crashing 1`] = `
<Text
  meta={Object {}}
  style={
    Object {
      "width": "100%",
    }
  }
  type="text"
>
  <div
    style={
      Object {
        "width": "100%",
      }
    }
  >
    <Base
      autoComplete="off"
      autoFocus={false}
      data-qa="base-input-element"
      data-qa-label="base-input-element-label"
      error={false}
      type="text"
    >
      <FloatingLabelWrapper
        data-qa-label="base-input-element-label"
        error={false}
      >
        <div
          className="FloatingLabelWrapper GenericInput "
          style={
            Object {
              "color": "hsl(0, 2%, 12%)",
              "cursor": "text",
              "position": "relative",
            }
          }
        >
          <div
            className="inputWrapper  bd-radius"
          >
            <input
              aria-invalid={false}
              autoComplete="off"
              autoFocus={false}
              data-qa="base-input-element"
              data-qa-label="base-input-element-label"
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              style={
                Object {
                  "color": "hsl(0, 2%, 12%)",
                  "cursor": "text",
                  "textAlign": undefined,
                }
              }
              type="text"
            />
            <span
              className="bar "
            />
          </div>
        </div>
      </FloatingLabelWrapper>
    </Base>
    <Fade>
      <Transition
        entered={
          Object {
            "opacity": 1,
          }
        }
        entering={
          Object {
            "opacity": 0,
          }
        }
        show={true}
        start={
          Object {
            "opacity": 0,
          }
        }
        transitionProperty="opacity"
      >
        <Transition
          appear={true}
          enter={true}
          exit={true}
          in={true}
          mountOnEnter={true}
          onEnter={[Function]}
          onEntered={[Function]}
          onEntering={[Function]}
          onExit={[Function]}
          onExited={[Function]}
          onExiting={[Function]}
          timeout={300}
          unmountOnExit={true}
        >
          <Component
            error={true}
            key="0/.0"
            style={
              Object {
                "bottom": -15,
                "left": 1,
                "opacity": 0,
                "position": "absolute",
                "transition": "all 300ms ease-in-out",
                "transitionProperty": "opacity",
              }
            }
          />
        </Transition>
      </Transition>
    </Fade>
  </div>
</Text>
`;
